src/helpViewer/helpProvider.ts

```diff
@@ -34,9 +34,12 @@ export class HelpProvider {
         this.port = this.launchRHelpServer(); // is a promise for now!
     }
 
-    public refresh(): void {
-        kill(this.cp.pid); // more reliable than cp.kill (?)
+    public async refresh(): Promise<void> {
+        if(!this.cp.exitCode){
```

Comment:
This check might not work as expected. I kill the help server process and here is what `cp` looks like when I try to restart the help server while debugging:

<img width="554" alt="image" src="https://user-images.githubusercontent.com/4662568/143774371-c3b45a2e-c4af-4e99-b39b-dadb44236bd4.png">

Reply:
Does the latest push work reliably for you?

Reply:
Thanks. The latest push is much more reliable.